project/AkkaDisciplinePlugin.scala

```diff
@@ -63,17 +63,51 @@ object AkkaDisciplinePlugin extends AutoPlugin {
     "akka-stream-tests-tck",
     "akka-testkit")
 
-  lazy val silencerSettings = {
-    val silencerVersion = "1.7.1"
-    val libs = Seq(
-      compilerPlugin(("com.github.ghik" %% "silencer-plugin" % silencerVersion).cross(CrossVersion.patch)),
-      ("com.github.ghik" %% "silencer-lib" % silencerVersion % Provided).cross(CrossVersion.patch))
-    Seq(libraryDependencies ++= (if (autoScalaLibrary.value) libs else Nil))
+  lazy val nowarnSettings = {
+    Dependencies.getScalaVersion() match {
+      // TODO: remove this when upgrading to Scala 2.12.13
+      // https://github.com/scala/scala/pull/9248
+      case twoTwelve if twoTwelve.startsWith("2.12") =>
+        val silencerVersion = "1.7.1"
+        val libs = Seq(
+          compilerPlugin(("com.github.ghik" %% "silencer-plugin" % silencerVersion).cross(CrossVersion.patch)),
+          ("com.github.ghik" %% "silencer-lib" % silencerVersion % Provided).cross(CrossVersion.patch))
+        Seq(libraryDependencies ++= (if (autoScalaLibrary.value) libs else Nil))
+      case _ =>
+        Seq(scalacOptions ++= Seq("-nowarn", "-Wconf:any:e"))
```

Comment:
I have tried a bunch of patterns, but I have troubles and I might be falling into bugs.
E.g.
With this configuration: `sbt -Dakka.build.scalaVersion=2.13 akka-actor/test:compile` result in 12 errors in `akka-actors` all in:
`@nowarn annotation does not suppress any warnings`

The full scalacConfiguration is:
```
akka > show akka-actor/test:scalacOptions
[info] * -nowarn
[info] * -Wconf:any:e
[info] * -Xfatal-warnings
[info] * -Ywarn-extra-implicit
[info] * -Ywarn-unused:_
[info] * -Xlint
[info] * -deprecation
[info] * -encoding
[info] * UTF-8
[info] * -feature
[info] * -unchecked
[info] * -language:higherKinds
[info] * -target:jvm-1.8
```
It would be nice to get help from someone like @lrytz 🙏

Reply:
👋 the compiler treats `-Wconf` before `-nowarn`. Using both `-nowarn -Wconf:any:e` first promotes all warnings to be errors, so the messages are issued using `reporter.error`, and `-nowarn` has no effect anymore as it's checked in `reporter.warning`.

`@nowarn annotation does not suppress any warnings` is a warning enabled by `-Xlint`, it means that a `@nowarn` annotation is redundant, i.e., doesn't silence anything. It is reported as an error here because of `-Wconf:any:e`.

Does that help..?

Reply:
Indeed, thanks a lot @lrytz I found my way out!
https://github.com/akka/akka/pull/29932/commits/1437da6bc33e8fc6eafeb6d33a0fa60bbd79e6ec
https://github.com/akka/akka/pull/29932/checks?check_run_id=1669212181